examples/hello-world/src/test-api.js

```diff
@@ -45,22 +45,22 @@ export default class TestApi extends FreshDataApi {
 		},
 	}
 
-	static endpoints = {
-		greetings: {
-			read: ( methods, endpointPath, params ) => {
-				const { get } = methods;
-				const fullEndpointPath = [ 'greetings', ...endpointPath ];
-				const value = get( fullEndpointPath )( params );
-				return value;
-			},
-		},
+	static operations = {
+		read: ( { get } ) => ( resourceNames ) => {
+			return resourceNames.reduce( ( requests, name ) => {
+				if ( 'greetings' === name ) {
+					requests[ name ] = get( [ name ] )();
+				}
```

Comment:
this is something that isn't where I'd like it with `http-data`. right now the call to `requestHttpData()` must specify the calling action for fetching the data. instead I'd rather have it register a way to get data of a given id and henceforth allow fetching via a call to `get( id )`

that introduces complexity you address here and I'm not sure how best to pull it into `http-data`, but thanks for this!

Reply:
I decided to take some inspiration from redux reducers here. Operations can be combined and composed, and a single operation can handle just one resourceName, or many. This becomes more evident in API like in WP REST API where you can pass in `includes=24,25,26,27,28`. This makes it possible to fulfill an operation on multiple resourceNames with one API call, which is one of the goals I was looking for.

In addition, it also allows the operation to save the data in custom ways. An example of this is fetching a page of posts, then storing each post under its own resourceName, and then storing the page as a simple list of IDs. Then the corresponding page selector can grab the page as a list of ids, then map those ids to the actual items for the page. The advantage of this is, the post data can be shared amongst multiple resource requests, such as page requests, search requests, or individual post requests. This also allows the same data to be updated after a mutating operation like an update, and then nothing has to be done to update the data on the page afterwards.